src/Framework/Analyzer/test/MinimalActions/DetectMismatchedParameterOptionalityTest.cs

```diff
@@ -0,0 +1,407 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+using VerifyCS = Microsoft.AspNetCore.Analyzers.Testing.Utilities.CSharpCodeFixVerifier<
+    Microsoft.AspNetCore.Analyzers.DelegateEndpoints.DelegateEndpointAnalyzer,
+    Microsoft.AspNetCore.Analyzers.DelegateEndpoints.DelegateEndpointFixer>;
+
+namespace Microsoft.AspNetCore.Analyzers.DelegateEndpoints;
+
+public partial class DetectMismatchedParameterOptionalityTest
+{
+    [Fact]
+    public async Task MatchingRequiredOptionality_CanBeFixed()
+    {
+        var source = @"
+#nullable enable
+using Microsoft.AspNetCore.Builder;
+
+class Program
+{
+    static void Main(string[] args)
+    {
+        var app = WebApplication.Create();
+        app.MapGet(""/hello/{name?}"", (string name) => $""Hello {name}"");
+    }
+}";
+        var fixedSource = @"
+#nullable enable
+using Microsoft.AspNetCore.Builder;
+
+class Program
```

Comment:
We can. I was running into some issues with CS8805 errors when using top-level programs. Turns out that is because the test infra assumes the projects represent a library by default. Changing the output kind to an exe allows us to validate top-level programs.